worksheet.rb

```diff
@@ -23,12 +45,150 @@
 # into this data structure, such as "DR0004"
 # and "3rd Feb 2016" and "RD0022"
 
+drivers = {
+
+    DR0001: [
+         {
+            rider_id: "RD0003",
+            date: "3rd Feb 2016",
+            cost: 10,
+            rating: 3
+        },
+        {
+            rider_id: "RD0015",
+            date: "3rd Feb 2016",
+            cost: 30,
+            rating: 4
+        },
+        {
+            rider_id: "RD0003",
+            date: "5th Feb 2016",
+            cost: 45,
+            rating: 2
+        },
+
+  ],
+
+    DR0002: [
+         {
+            rider_id: "RD0073",
+            date: "3rd Feb 2016",
+            cost: 25,
+            rating: 5
+        },
+        {
+            rider_id: "RD0003",
+            date: "4th Feb 2016",
+            cost: 15,
+            rating: 1
+        },
+        {
+            rider_id: "RD0003",
+            date: "5th Feb 2016",
+            cost: 35,
+            rating: 3
+          },
+    ],
+
+    DDR0003: [
+        {
+            rider_id: "RD0066",
+            date: "4th Feb 2016",
+            cost: 5,
+            rating: 5
+        },
+        {
+            rider_id: "RD0003",
+            date: "5th Feb 2016",
+            cost: 50,
+            rating: 2
+        },
+    ],
+
+    DR0004: [
+        {
+            rider_id: "RD0022", 
+            date: "3rd Feb 2016",
+            cost: 5,
+            rating: 5
+        },
+        {
+            rider_id: "RD0022",
+            date: "4th Feb 2016",
+            cost: 10,
+            rating: 4
+        },
+        {
+            rider_id: "RD0073",
+            date: "5th Feb 2016",
+            cost: 20,
+            rating: 5
+        }
+    ]
+  }
+
+
+
 ########################################################
 # Step 4: Total Driver's Earnings and Number of Rides
 
 # Use an iteration blocks to print the following answers:
 # - the number of rides each driver has given
+
+drivers.each do |driver, trips|
+  puts "#{driver}: #{trips.length}"
+end
+
 # - the total amount of money each driver has made
+
+total_earned = []
+drivers.each do |driver, trips|
+  total_costs = trips.sum do |item| 
+      item[:cost]
+  end
+  total_earned << {driver => total_costs}
+end
+
+puts total_earned
+
 # - the average rating for each driver
+
+avg_ratings = []
+drivers.each do |driver, trips|
+ total_rating = trips.sum do |item|
+    item[:rating]  
+ end
+  avg = total_rating/trips.length
```

Comment:
So the averages you are giving are rounding to the nearest whole number because of how you are doing the math. remember, in a situation like this, a `.to_f` on either the `length` or `total_ratings` would go a long way to preserving data.